### ISSUE TITLE
Speculative prerendering

### ISSUE THREAD

**Issue by felixarntz** [WordPress/performance#908]
This issue provides an overview of the work related to the "Speculation Rules" module. See the `feature/speculationrules` branch for development code. **Update:** After #946, development should continue in `feature/modules-to-plugins` and eventually `trunk` once that branch is merged into it.

* [x] #897 
* [x] #905
* [x] #906
* [x] #907
* [x] #909
